### ISSUE TITLE
How to Assign IP Address to Hyper V Hosted VM while VM Creation

### ISSUE THREAD

**Issue by rsganta** [PowerShell/xNetworking#278]
I would like to assign IP Address to the OnPrem hyper V VM using DSC automation.  As I do want to run this to be happen in dynamic way. Then I thought of installing MMA on the VM to utilize the Hybridworker option. However since this VM does not have any IP, unable to configure MMA successfully. Second option tried to run LCM on this machine but it failed because of no IP Address. Third option is to assign the IP while creation of the VM  . As I am using image and I want to create 10 VMs with valid IPAddresses in the domain. what would be the best way to assign the IP address dynamically to the Onprem Hyper V vm .  Any insights would be greatly appreciated

**Comment by PlagueHO** [comment]
Hi There @rsganta - are you using a DSC pull server in this environment? If not then you still should be able to cause the LCM on the target nodes to apply a DSC config. Even when an IP Address can't be set statically or obtained from a DHCP an automatic private IP Address (APIPA) is usually assigned on Windows 10 - which will allow you to apply the DSC config to the LCM in push mode.

Pull mode is another story - you would need a network connection for that.

Can you clarify what is stopping you applying the config to the VM in push mode?

**Comment by rsganta** [comment]
Here is the flow as per my understanding.
1. I create a vm using powershell runbook (discussing on a different thread to rename before it starts)
2. Once It got renamed, my LCM runs to create a .MOF file.
3. In parallel to step 2, I have my configuration uploaded to azure using the renamed machined name and compile the job and assign node configuration.
4. My configuration has domainjoin,ipaddress,subnetmask and dns etc.
The problem with step 4 is, Since it is a private domain and assigning static IP. However when the IP got assigned  , network connection goes to unidentified network.   If I leave just with domain join(which has already valid IP from DHCP ), then rest of the things work.  But the problem is I fail to assign static ip, gateway,subnetmask and dns...
